File: FormalBench-data/FilteredData/translated_c/seed_726_per_category_10_653ade686f/OddBitSetNumber.c
#include <stdint.h>

int32_t oddBitSetNumber(int32_t n) {
    uint32_t value = (uint32_t)n;
    value |= ((uint32_t)n & UINT32_C(0xAAAAAAAA)) >> 1;
    value |= ((uint32_t)n & UINT32_C(0xCCCCCCCC)) >> 2;
    value |= ((uint32_t)n & UINT32_C(0xF0F0F0F0)) >> 4;
    value |= ((uint32_t)n & UINT32_C(0xFF00FF00)) >> 8;
    value |= ((uint32_t)n & UINT32_C(0xFFFF0000)) >> 16;
    return (int32_t)value;
}

File: FormalBench-data/FilteredData/selected_java/seed_726_per_category_10_653ade686f/OddBitSetNumber.java
import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;


class OddBitSetNumber {
    
    public static int oddBitSetNumber(int n) {
        int n_ = n & 0xFFFFFFFF;
        n_ |= (n & 0xAAAAAAAA) >>> 1;
        n_ |= (n & 0xCCCCCCCC) >>> 2;
        n_ |= (n & 0xF0F0F0F0) >>> 4;
        n_ |= (n & 0xFF00FF00) >>> 8;
        n_ |= (n & 0xFFFF0000) >>> 16;
        return n_;
    }
}

